crates/viewer/re_space_view_spatial/src/mesh_loader.rs

```diff
@@ -37,41 +45,58 @@ impl LoadedMesh {
     pub fn load_asset3d_parts(
         name: String,
         media_type: &MediaType,
-        bytes: &[u8],
+        asset3d: &Asset3D,
         render_ctx: &RenderContext,
+        texture_key: u64,
     ) -> anyhow::Result<Self> {
         re_tracing::profile_function!();
 
+        let bytes = asset3d.blob.as_slice();
+
         let mesh_instances = match media_type.as_str() {
             MediaType::GLTF | MediaType::GLB => {
                 re_renderer::importer::gltf::load_gltf_from_buffer(&name, bytes, render_ctx)?
             }
             MediaType::OBJ => re_renderer::importer::obj::load_obj_from_buffer(bytes, render_ctx)?,
-            MediaType::STL => re_renderer::importer::stl::load_stl_from_buffer(bytes, render_ctx)?,
+            MediaType::STL => {
+                re_renderer::importer::stl::load_stl_from_buffer(bytes, render_ctx, texture_key)?
+            }
             _ => anyhow::bail!("{media_type} files are not supported"),
         };
 
-        let bbox = re_renderer::importer::calculate_bounding_box(&mesh_instances);
+        let prev_mesh_instances = mesh_instances.clone();
+        let mesh_instances = add_albedo_factor_to_mesh(mesh_instances, asset3d, render_ctx);
```

Comment:
I don't see why `mesh_instances` is cloned. You should instead modify it in-place, that would be a lot cheaper & easier to follow.

Edit: You're actually cloning it 2x now since `add_albedo_factor_to_mesh` creates another copy of them. Please avoid this!